packages/common/src/lib/notify/notify-builder.js

```diff
@@ -1,184 +1,212 @@
 const logger = require('../logger');
 const { createNotifyClient } = require('./notify-factory');
 
-function NotifyBuilder() {
-  function setPrivateNotifyClient(notifyClient) {
-    logger.debug({ notifyClient }, 'Setting notify client');
-    this.notifyClient = notifyClient;
-  }
+module.exports = {
+  /**
+   * Resets all the data on the object. Useful for cleaning the client between emails or tests.
+   *
+   * @returns {*}
+   */
+  reset() {
+    logger.info('Resetting the notify client');
+    this.notifyClient = undefined;
+    this.templateId = undefined;
+    this.templatePersonalisation = undefined;
+    this.destinationEmail = undefined;
+    this.emailReplyToId = undefined;
+    this.reference = undefined;
+
+    return this;
+  },
```

Comment:
I'm honestly not sure how I feel about this. 

My personal opinion was that we should not use a single instance of the notify client to send multiple emails as it leads to the risk that this function has been added to avoid.

This now relies on the developer remembering to `reset` between uses, and of greater concern, requires future development to ensure this method is kept updated. 

imho, it was preferable to explicitly build a new instance of the notify wrapper per email dispatch, to avoid the risk of retaining one or more pieces of something else's data.

Reply:
I agree with this. There should be a new instance. Does that have an impact upon state?

Reply:
Yes, the difference is if a new instance is created per send, we get a clean state by virtue of construction. This was the previous implementation.

In this PR's implementation, we get a clean state on instantiation, but the state is only wiped if the consumer explicitly calls `reset()`. 

I can live with this, but I'm not sold on this being an improvement on what we had. 

That said, the other approach was not perfect by any means.